libusb/os/darwin_usb.c

```diff
@@ -2139,6 +2143,11 @@ static int darwin_reenumerate_device (struct libusb_device_handle *dev_handle, b
 
   struct libusb_context *ctx = HANDLE_CTX (dev_handle);
 
+  if (!dpriv || !dpriv->device) {
```

Comment:
The check for `!dpriv` is not necessary, dpriv is already dereferenced 10 lines before, so would have crashed already at this point.